pkg/plugins/resources/k8s/native/api/v1alpha1/gateway_instance.go

```diff
@@ -0,0 +1,65 @@
+package v1alpha1
+
+import (
+	kube_core "k8s.io/api/core/v1"
+	kube_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:root=true
+
+// GatewayInstance represents a managed instance of a dataplane proxy for a Kuma
+// Gateway.
+type GatewayInstance struct {
+	kube_meta.TypeMeta   `json:",inline"`
+	kube_meta.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec GatewayInstanceSpec `json:"spec,omitempty"`
+}
```

Comment:
Should add `Status` as a subresource.

Reply:
How about this as a starting proposal?

```Go
type GatewayInstanceStatus struct {
	// LoadBalancer contains the current status of the load-balancer,
	// if one is present.
	// +optional
	LoadBalancer *LoadBalancerStatus `json:"loadBalancer,omitempty"

	// Current service state
	// +optional
	// +patchMergeKey=type
	// +patchStrategy=merge
	// +listType=map
	// +listMapKey=type
	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"
}
```

For Conditions, maybe we can get away with just the conventional `Ready`, which is True if both the deployment and service are present and up to date?

Reply:
Yeah I think that makes sense.